test/hooks.test.js

```diff
@@ -259,6 +259,98 @@ test('preHandler hook should support encapsulation / 5', t => {
   })
 })
 
+test('onRoute hook should be called / 1', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  fastify.register((instance, opts, next) => {
+    instance.addHook('onRoute', () => {
+      t.pass()
+    })
+    instance.get('/', opts, function (req, reply) {
+      reply.send()
+    })
+    next()
+  })
+
+  fastify.ready(err => {
+    t.error(err)
+  })
+})
+
+test('onRoute hook should be called / 2', t => {
+  t.plan(3)
+  const fastify = Fastify()
+  fastify.addHook('onRoute', (route) => {
+    t.pass()
+  })
+
+  fastify.register((instance, opts, next) => {
+    instance.addHook('onRoute', (route) => {
+      t.pass()
+    })
+    instance.get('/', opts, function (req, reply) {
+      reply.send()
+    })
+    next()
+  })
+
+  fastify.ready(err => {
+    t.error(err)
+  })
+})
+
+test('onRoute should keep the context', t => {
+  t.plan(4)
+  const fastify = Fastify()
+  fastify.register(plugin)
```

Comment:
could you just pass the function `plugin` directly without defining it after?

Reply:
Done